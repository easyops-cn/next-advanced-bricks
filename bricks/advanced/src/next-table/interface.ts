import type { ColumnType, TablePaginationConfig } from "antd/es/table";
import type {
  SortOrder,
  ExpandableConfig as TableExpandableConfig,
  TableRowSelection,
} from "antd/es/table/interface.js";
import type { UseBrickConf } from "@next-core/types";
import { CSSProperties } from "react";
import type { TableProps } from "antd/es/table";

export interface NextTableProps {
  rowKey?: string;
  columns?: Column[];
  cell?: CellConfig;
  dataSource?: DataSource;
  frontSearch?: boolean;
  pagination?: PaginationType;
  loading?: boolean;
  multiSort?: boolean;
  sort?: Sort | Sort[];
  rowSelection?: RowSelectionType;
  selectedRowKeys?: (string | number)[];
  hiddenColumns?: (string | number)[];
  expandable?: ExpandableType;
  expandedRowKeys?: (string | number)[];
  childrenColumnName?: string;
  rowDraggable?: boolean;
  searchFields?: (string | string[])[];
  size?: TableProps<RecordType>["size"];
  showHeader?: boolean;
  bordered?: boolean;
  scrollConfig?: TableScroll;
  optimizedColumns?: (string | number)[];
  themeVariant?: "default" | "elevo";
}

export type TableScroll = TableProps<RecordType>["scroll"] | false;

export type RecordType = Record<string, any>;

export interface Column
  extends Pick<
    ColumnType<RecordType>,
    | "align"
    | "colSpan"
    | "dataIndex"
    | "ellipsis"
    | "fixed"
    | "rowScope"
    | "showSorterTooltip"
    | "sortDirections"
    | "title"
    | "width"
  > {
  key?: string | number;
  /** 表头自定义展示构件 */
  headerBrick?: WithUseBrick;
  /** 内容自定义展示构件 */
  useBrick?: UseBrickConf;
  /** 记录表格列合并的值的 key */
  cellColSpanKey?: string;
  /** 记录表格行合并的值的 key */
  cellRowSpanKey?: string;
  /** 是否可排序 */
  sortable?: boolean;
  /** 前端搜索时，多列排序优先级，数字越大优先级越高 */
  sortPriority?: number;
  /** 垂直对齐方式 */
  verticalAlign?: CSSProperties["verticalAlign"];
  /** 单元格样式 */
  cellStyle?: CSSProperties;
  /** 头部单元格样式 */
  headerStyle?: CSSProperties;
  cellStatus?: CellStatus;
}

export interface CellStatus {
  dataIndex?: string;
  mapping: CellStatusMap[];
}

export interface CellStatusMap {
  value: unknown;
  leftBorderColor: string;
}

export interface CellConfig extends WithUseBrick {
  header?: WithUseBrick;
}

export interface WithUseBrick {
  useBrick?: UseBrickConf;
}

export interface DataSource {
  list?: RecordType[];
  total?: number;
  page?: number;
  pageSize?: number;
}

interface PaginationConfig
  extends Pick<
    TablePaginationConfig,
    | "disabled"
    | "hideOnSinglePage"
    | "pageSizeOptions"
    | "responsive"
    | "showLessItems"
    | "showQuickJumper"
    | "showSizeChanger"
    | "showTitle"
    | "simple"
    | "size"
  > {
  showTotal?: boolean;
}

export type PaginationType = false | PaginationConfig | undefined;

interface RowSelectionConfig
  extends Pick<
    TableRowSelection<RecordType>,
    | "columnWidth"
    | "fixed"
    | "hideSelectAll"
    | "preserveSelectedRowKeys"
    | "type"
    | "checkStrictly"
  > {
  showSelectInfo?: boolean;
  rowDisabled?: string | boolean;
}

export type RowSelectionType = boolean | RowSelectionConfig | undefined;

interface ExpandableConfig
  extends Pick<
    TableExpandableConfig<RecordType>,
    | "columnWidth"
    | "expandRowByClick"
    | "defaultExpandAllRows"
    | "fixed"
    | "showExpandColumn"
  > {
  expandIconBrick?: {
    useBrick: UseBrickConf;
  };
  expandedRowBrick?: {
    useBrick: UseBrickConf;
  };
  rowExpandable?: string | boolean;
}

export type ExpandableType = boolean | ExpandableConfig | undefined;

export interface Sort {
  columnKey?: string | number;
  order?: SortOrder;
}
