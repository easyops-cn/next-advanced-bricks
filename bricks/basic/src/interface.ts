import type { GeneralIconProps } from "@next-bricks/icons/general-icon";
/* ============== Common Type Start ============== */

export type Target = "_self" | "_blank" | "_parent" | "_top";

export type UIType = "default" | "dashboard";

export type Shape = "round" | "circle";

/* ============== Common Type Start ============== */

/* ============== Button Type Start ============== */

export type ButtonType =
  | "primary"
  | "default"
  | "dashed"
  | "ghost"
  | "link"
  | "text";

export type ComponentSize = "large" | "medium" | "small" | "xs";

/* ============== Button Type End ============== */

/* ============== Link Type Start ============== */

export type LinkType = "link" | "text" | "plain";

/* ============== Link Type End ============== */

/** ============= Popover Type Start ============== */
export type TriggerEvent = "click" | "hover";
/** ============= Popover Type End ============== */
