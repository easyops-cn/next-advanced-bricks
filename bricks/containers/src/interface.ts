/* ============== Common Type Start ============== */

export type Target = "_self" | "_blank" | "_parent" | "_top";

export type UIType = "default" | "dashboard";

export type Shape = "round" | "circle";

export type ComponentSize = "large" | "middle" | "small" | "xs";

/* ============== Common Type End ============== */

/* ============== Drawer Type Start ============== */

export type Placement = "left" | "right" | "top" | "bottom";

/* ============== Drawer Type Start ============== */
