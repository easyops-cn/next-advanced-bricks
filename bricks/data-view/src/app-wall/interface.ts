import React from "react";
import {Group} from "@tweenjs/tween.js";
import { Vector3Tuple } from "three";
interface TrapezoidalObjectData {
     width: number;
     height: number;
     point: Vector3Tuple
}
export interface TrapezoidalObjectProps {
     // objectRef?: React.RefObject<CSS3DObject>;
     trapezoidalTweenRef: React.RefObject<Group>;
     objectData: TrapezoidalObjectData;
     leftBtnName?: string | undefined;
     rightBtnName?: string | undefined;
     leftOnClick?: () => void;
     rightOnClick?: () => void;

 }
