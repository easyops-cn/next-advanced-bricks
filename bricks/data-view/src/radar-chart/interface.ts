type Color = React.CSSProperties["color"];
export type Colors = Color[][];
type NumArr = number[];
export type Axis = {
  x: number;
  y: number;
};
export interface DrawRadarBgOption {
  layer: number;
  n: number;
  r: number;
  ratio: number;
  origin: number[];
  lineWidth?: number;
}
export interface DrawPolygonOptions extends Omit<DrawRadarBgOption, "layer"> {
  fillStyle?: Color | Colors;
  strokeStyle?: string;
}

export interface GradientPolygon extends DrawPolygonOptions {}
export interface DataLineOptions {
  dataPoints: Axis[];
  strokeStyle?: Color;
  lineWidth?: number;
}
export interface DataCircleOptions extends DataLineOptions {
  r?: number;
  fillStyle?: Color;
}
export interface CircleOptions extends Axis {
  originX: number;
  originY: number;
  r: number;
  lineWidth?: number;
  strokeStyle?: Color;
  fillStyle?: Color;
}
export interface DataFillProps {
  dataPoints: Axis[];
  fillStyle?: Color;
}
export interface DataPolyProps {
  dataLineOptions: DataLineOptions;
  dataFillOptions: DataFillProps;
  dataCircleOptions: DataCircleOptions;
}
export interface BaseConfig {
  n: number;
  dataRadiusOfPercent: NumArr;
  dataRadius: NumArr;
  angleArr: NumArr;
  tooltipsContentArr: NumArr;
  radius: number;
  originDataSource: Data[];
}
export interface UserConfig {
  dataFill: DataFill;
  dataCircle: DataCircle;
  dataLine: DataLine;
}
export interface Data {
  value: number;
  maxValue: number;
  name?: string;
  color?: Color;
  percentValue?: number | string;
}
export type DataFill = {
  fillStyle?: Color;
};
export type DataLine = {
  lineWidth?: number;
  strokeStyle?: Color;
};
export type DataCircle = Omit<DataCircleOptions, "dataPoints">;
export interface RadarProps {
  dataSource: Data[];
  height?: number;
  width?: number;
  radius?: number;
  scale?: number;
  value?: number | string;
  dataFill?: DataFill;
  dataCircle?: DataCircle;
  dataLine?: DataLine;
}
