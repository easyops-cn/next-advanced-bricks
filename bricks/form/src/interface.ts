/* ============== Common Type Start ============== */

export type Target = "_self" | "_blank" | "_parent" | "_top";

export type UIType = "default" | "dashboard";

export type Shape = "round" | "circle";

/* ============== Common Type Start ============== */

/* ============== Button Type Start ============== */

export type ButtonType =
  | "primary"
  | "default"
  | "dashed"
  | "ghost"
  | "link"
  | "text";

export type ComponentSize = "large" | "medium" | "small" | "xs";

/* ============== Button Type End ============== */

/* ============== Input Type Start ============== */

export type InputType =
  | "button"
  | "checkbox"
  | "color"
  | "date"
  | "datetime-local"
  | "email"
  | "file"
  | "hidden"
  | "image"
  | "month"
  | "number"
  | "password"
  | "radio"
  | "range"
  | "reset"
  | "search"
  | "submit"
  | "tel"
  | "text"
  | "time"
  | "url"
  | "week";

/* ============== Input Type End ============== */

/* ============== Input Type Start ============== */

export type Layout = "horizontal" | "vertical" | "inline";

/* ============== Input Type End ============== */

/* ============== Radio Type Start ============== */

export type RadioType =
  | "button"
  | "default"
  | "icon"
  | "icon-circle"
  | "icon-square"
  | "custom";
export interface GeneralComplexOption<T = string | number | boolean> {
  label: string;
  value: T;
  [key: string]: any;
}

export declare type GeneralOption =
  | string
  | number
  | boolean
  | GeneralComplexOption
  | Record<string, any>;

export interface GeneralGroupOption {
  label: string;
  options: GeneralComplexOption | GeneralOption;
}

export declare type RadioGroupButtonStyle = "outline" | "solid";

/* ============== Radio Type End ============== */
