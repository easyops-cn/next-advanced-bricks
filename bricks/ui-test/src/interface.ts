import type { GeneralIconProps } from "@next-bricks/icons/general-icon";

export enum NodeType {
  Suite = "suite",
  Block = "block",
  Command = "command",
}

export interface NodeItem {
  type: NodeType;
  children?: NodeItem[];
  label?: string;
  name: string;
  params?: any[];
}

export interface CommandParam {
  label: string;
  type: CommandParamType;
  description?: string;
  required?: boolean;
  default?: unknown;
  enum?: (string | number | EnumItem)[];
}

export interface CommandDoc {
  name: string;
  category: CommandCategory;
  from: CommandFrom;
  chain: CommandChain;
  description?: string;
  params?: CommandParam[];
  overloads?: CommandOverload[];
  keywords?: string[];
  icon?: CommandIcon;
}

export interface CommandOverload {
  label: string;
  params?: CommandParam[];
}

export type CommandCategory = "query" | "action" | "assertion" | "other";
export type CommandFrom = "cypress" | "third-party" | "custom";
export type CommandChain = "parent" | "child" | "dual";

export type CommandParamType =
  | "number"
  | "string"
  | "boolean"
  | "object"
  | "array"
  | "function"
  | "mixed";

export interface EnumItem {
  label: string;
  value: unknown;
}

export type CommandIcon = GeneralIconProps & {
  color?: string;
};

export interface NodeGraphData {
  name: string;
  label?: string;
  instanceId: string;
  params?: any;
  type: NodeType;
  sort?: number;
}

export interface TreeNodeItemData extends NodeGraphData {
  isChainChild?: boolean;
  isSortChange?: boolean;
  displayLabel?: string;
  parent?: NodeGraphData;
  children?: NodeGraphData[];
}

export interface TestTreeData {
  key: string;
  name: string;
  icon: CommandIcon;
  data: TreeNodeItemData;
  children: TestTreeData[];
  labelColor?: string;
}
