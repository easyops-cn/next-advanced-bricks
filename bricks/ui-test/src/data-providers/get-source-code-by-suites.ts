import { createProviderClass } from "@next-core/utils/general";
import { parseSourceCode } from "../utils/parseSuiteAst";
import { NodeItem, TransformOptions } from "../interface";

interface SourceResult {
  name: string;
  source: string;
}

export async function getSourceCodeBySuites(
  suiteDataList: NodeItem[],
  transformOptions?: TransformOptions
): Promise<SourceResult[]> {
  const sourceList = [] as SourceResult[];
  suiteDataList?.forEach((suiteData) => {
    const code = parseSourceCode(suiteData, transformOptions);

    sourceList.push({
      name: suiteData.name,
      source: code as string,
    });
  });

  return sourceList;
}

customElements.define(
  "ui-test.get-source-code-by-suites",
  createProviderClass(getSourceCodeBySuites)
);
