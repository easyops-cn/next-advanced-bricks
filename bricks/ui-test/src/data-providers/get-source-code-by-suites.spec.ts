import { describe, test, expect } from "@jest/globals";
import { getSourceCodeBySuites } from "./get-source-code-by-suites.js";
import { NodeItem } from "../interface.js";

describe("getSourceCodeBySuites", () => {
  test("should work", async () => {
    const suiteData = [
      {
        type: "suite",
        name: "new",
        label: "new",
        children: [
          {
            type: "block",
            name: "describe",
            label: "new",
            children: [
              {
                type: "block",
                name: "beforeEach",
                children: [
                  {
                    type: "command",
                    name: "setLanguage",
                    label: "test",
                    params: ["en"],
                    children: [],
                  },
                  {
                    type: "command",
                    name: "login",
                    children: [],
                  },
                ],
              },
              {
                type: "block",
                name: "it",
                label: "test",
                children: [
                  {
                    type: "command",
                    name: "get",
                    params: ["test"],
                    children: [
                      {
                        type: "command",
                        name: "should:be.empty",
                        children: [],
                      },
                      {
                        type: "command",
                        name: "contains",
                        params: ["test"],
                        children: [],
                      },
                    ],
                  },
                  {
                    type: "command",
                    name: "get",
                    params: ["test"],
                    children: [],
                  },
                ],
              },
            ],
          },
        ],
      },
    ] as NodeItem[];
    expect(await getSourceCodeBySuites(suiteData)).toMatchSnapshot();
  });
});
