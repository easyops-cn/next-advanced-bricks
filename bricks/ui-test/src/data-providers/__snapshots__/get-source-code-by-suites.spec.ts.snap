// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSourceCodeBySuites should work 1`] = `
[
  {
    "name": "new",
    "source": "describe("new", () => {
  beforeEach(() => {
    cy.setLanguage("en");
    cy.login();
  });
  it("test", () => {
    cy.get("test").should("be.empty").contains("test");
    cy.get("test");
  });
});",
  },
]
`;
